internal/controller/driver_controller.go

```diff
@@ -756,6 +756,32 @@ func (r *driverReconcile) reconcileControllerPluginDeployment() error {
 								),
 							})
 						}
+						// ODF Snapshotter Sidecar Container
+						if r.driverType != NfsDriverType && snPolicy == csiv1a1.VolumeGroupSnapshotPolicy && deployPrivateSnapshotSidecar == "true" {
```

Comment:
we dont need `snPolicy == csiv1a1.VolumeGroupSnapshotPolicy` check, it can be removed

instead of comparing a string parse the bool and use it.